### PR TITLE
SupportBundles: Build tars in memory

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Build tar in memory for support bundle and store the result in KV store

### CODE REVIEW

File: pkg/services/supportbundles/supportbundlesimpl/store.go [hunk 1]
```diff
@@ -108,6 +108,8 @@ func (s *store) List() ([]supportbundles.Bundle, error) {
 			if err := json.NewDecoder(strings.NewReader(s)).Decode(&b); err != nil {
 				return nil, err
 			}
+
+			b.TarBytes = nil
```
Comment (Jguer) [inline pkg/services/supportbundles/supportbundlesimpl/store.go:112]: don't include the tar bytes in the list DTO

File: pkg/services/supportbundles/interface.go [hunk 2]
```diff
@@ -23,10 +23,10 @@ func (s State) String() string {
 type Bundle struct {
 	UID       string `json:"uid"`
 	State     State  `json:"state"`
-	FilePath  string `json:"filePath"`
 	Creator   string `json:"creator"`
 	CreatedAt int64  `json:"createdAt"`
 	ExpiresAt int64  `json:"expiresAt"`
+	TarBytes  []byte `json:"bytesTar,omitempty"`
```
Comment (kalleep) [inline pkg/services/supportbundles/interface.go:29]:
```suggestion
	TarBytes  []byte `json:"-"`
```

Comment (kalleep) [reply pkg/services/supportbundles/interface.go:29]: We never pass the tar with a bundle response

Comment (Jguer) [reply pkg/services/supportbundles/interface.go:29]: but if we do that we cannot extract it from the cache/kv :sweat_smile:

As an alternative I can embed a second struct inside, wdyt?

Comment (kalleep) [reply pkg/services/supportbundles/interface.go:29]: Oh yeah, that is right. Then we can just keep the tag I guess and make sure we are not passing it when listing.

**Review by kalleep — APPROVED** [review]

**Pull request merged**